### PR TITLE
Curl: Build curl-gnutls in the same package

### PR DESCRIPTION
**Pull request by ReillyBrogan**
**Summary**
- Build curl with brotli and IDN2 support
- Build the curl-gnutls package in the same package (resurrecting the effort by @livingsilver94)

### CODE REVIEW

File: packages/c/curl/package.yml [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+    - gnutls : GnuTLS version of curl
+    - gnutls-32bit : GnuTLS version of curl
+description:
+    - curl is a client to get files from servers using any of the supported protocols. The command is designed to work without user interaction or any kind of interactivity. curl offers a busload of useful tricks like proxy support, user authentication, ftp upload, HTTP post, file transfer resume and more.
+    - gnutls : This package provides binary compatibility only and enables the running of content expecting a GnuTLS version of curl. It is forbidden to build against it.
+    - gnutls-32bit : This package provides binary compatibility only and enables the running of content expecting a GnuTLS version of curl. It is forbidden to build against it.
 emul32     : yes
 summary    : Get a file from an HTTP, HTTPS or FTP server
 description: |
     curl is a client to get files from servers using any of the supported protocols. The command is designed to work without user interaction or any kind of interactivity.  curl offers a busload of useful tricks like proxy support, user authentication, ftp upload, HTTP post, file transfer resume and more.
```
Comment (EbonJaeger) [inline packages/c/curl/package.yml:24]: Should this have been deleted?

**Review by EbonJaeger — CHANGES_REQUESTED** [review]
Is `libgnutls` not being in `system.base` (because it's a dependency of `curl`) going to be a problem?

File: packages/c/curl/package.yml (hunk 1 above)
Comment (ReillyBrogan) [reply packages/c/curl/package.yml]: You saw nothing! [POCKET SAND](https://www.youtube.com/watch?v=QLpUq__iQqw)

**Comment by ReillyBrogan** [comment]
> Is `libgnutls` not being in `system.base` (because it's a dependency of `curl`) going to be a problem?

No, because it only ends up being a dependency of the curl-gnutls subpackage which is not in system.base.

**Comment by ReillyBrogan** [comment]
Hold please I fucked up the commits

**Review by EbonJaeger — APPROVED** [review]
LGTM, thanks!

**Pull request merged**